jmix-flowui/flowui/src/main/java/io/jmix/flowui/component/navigationmenubar/NavigationMenuBarItemConverter.java

```diff
@@ -0,0 +1,262 @@
+/*
+ * Copyright 2023 Haulmont.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.jmix.flowui.component.navigationmenubar;
+
+import com.google.common.base.Splitter;
+import com.google.common.base.Strings;
+import com.vaadin.flow.component.UI;
+import com.vaadin.flow.component.icon.Icon;
+import com.vaadin.flow.component.icon.VaadinIcon;
+import io.jmix.core.MessageTools;
+import io.jmix.flowui.menu.MenuConfig;
+import io.jmix.flowui.menu.MenuItem;
+import io.jmix.flowui.menu.MenuItemCommand;
+import io.jmix.flowui.menu.MenuItemCommands;
+import io.jmix.flowui.sys.UiAccessChecker;
+import io.jmix.flowui.view.View;
+import io.jmix.flowui.view.ViewController;
+import io.jmix.flowui.view.ViewInfo;
+import io.jmix.flowui.view.ViewRegistry;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.lang.Nullable;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Consumer;
+
+@Component("flowui_NavigationMenuBarItemConverter")
+public class NavigationMenuBarItemConverter {
```

Comment:
For ListMenu similar bean is called `xxBuilder`. Don't we want a similar name?

Reply:
It doesn't build a menu (as ListMenuBuilder does but it's not used anywhere in the framework), just converts menu item descriptors to menu items. Also, because it's not a builder, it's scope is singleton (ListMenuBuilder has prototype scope). I would not rename it to builder.